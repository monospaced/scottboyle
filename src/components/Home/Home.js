import DocumentMeta from "react-document-meta";
import marked from "marked";
import React from "react";

import "../Main/Main.css";
import Avatar from "../Avatar/Avatar";

const Home = ({ data: { about, description, subtitle, title } }) => {
  return (
    <main className="Main vcard">
      <DocumentMeta
        title={`${title} | ${subtitle}`}
        description={description}
      />
      <section>
        <Avatar />
        <div dangerouslySetInnerHTML={{ __html: marked.parse(about.trim()) }} />
      </section>
    </main>
  );
};

export default Home;
