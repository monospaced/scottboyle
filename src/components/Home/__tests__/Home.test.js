import React from "react";
import { shallow } from "enzyme";

import data from "../../../scripts/__mocks__/data.js";
import Home from "../Home.js";

describe("Home component", () => {
  const { about, description, subtitle, title } = data;

  it("should render correctly", () => {
    const props = {
      data: { about, description, subtitle, title },
    };
    const component = shallow(<Home {...props} />);
    expect(component).toMatchSnapshot();
  });
});
