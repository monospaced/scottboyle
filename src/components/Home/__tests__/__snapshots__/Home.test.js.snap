// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render correctly 1`] = `
<main
  className="Main vcard"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio"
  />
  <section>
    <Avatar />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p><span class=\\"fn\\">Scott Boyle</span> is a <span class=\\"category\\">web professional</span> <span class=\\"note\\">specialising in modern front-end development</span>. Scott currently lives in <span class=\\"adr\\"><span class=\\"locality\\">London</span> <span class=\\"country-name\\">UK</span></span> and has a company called <a href=\\"https://monospaced.com\\" class=\\"org\\">Monospaced&nbsp;Ltd</a>, a freelance web development consultancy working with regular clients in London and around the&nbsp;world.</p>
<h3 id=\\"view-source\\">View source</h3>
<p><a href=\\"https://github.com/monospaced/scottboyle\\" rel=\\"source\\">github.com/monospaced/scottboyle</a></p>
",
        }
      }
    />
  </section>
</main>
`;
